networkx/utils/backends.py

```diff
@@ -127,6 +127,20 @@ def _get_backends(group, *, load_and_call=False):
 backends = _get_backends("networkx.backends")
 backend_info = _get_backends("networkx.backend_info", load_and_call=True)
 
+config = {
+    # Get default configuration from environment variables at import time
+    "automatic_backends": [
+        x.strip()
+        for x in os.environ.get("NETWORKX_AUTOMATIC_BACKENDS", "").split(",")
+        if x.strip()
+    ],
+    # Initialize default configuration for backends
+    "backends": {
+        backend: info.get("default_config", {})
+        for backend, info in backend_info.items()
+    },
+}
```

Comment:
This may be for a separate PR but I think the config entry "automatic backends" could be named better (same for the corresponding env var), especially when seen next to the config entry here for "backends".

We should discuss this a bit more, but I think "backend priority" (`config["backend_priority"]`, `NETWORKX_BACKEND_PRIORITY`) would be a better name.

Reply:
Sounds like a good thing to discuss :+1:

Reply:
I do think maybe we can think a bit more about the schema for the config dictionary, but for now let's go with this!